lib/bundler/lockfile_generator.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+module Bundler
+  class LockfileGenerator
+    attr_reader :definition
+    attr_reader :out
+
+    # @private
+    def initialize(definition)
+      @definition = definition
+      @out = String.new
+    end
+
+    def self.generate(definition)
+      new(definition).generate!
+    end
+
+    def generate!
+      add_sources
+      add_platforms
+      add_dependencies
+      add_locked_ruby_version
+      add_bundled_with
+
+      out
+    end
+
+  private
+
+    def add_sources
+      definition.send(:sources).lock_sources.each_with_index do |source, idx|
```

Comment:
its not public now and I didn't want to change access control. Eventually, I think all the definition properties that are accessed should be passed into the `LockfileGenerator` but considered it as a second step